src/umpire/DeviceAllocator.cpp

```diff
@@ -79,9 +79,9 @@ __device__ void* DeviceAllocator::allocate(size_t size)
   return static_cast<void*>(m_ptr + counter);
 }
 
-__host__ __device__ size_t DeviceAllocator::getID()
+__host__ __device__ int DeviceAllocator::getID()
 {
-  return m_id;
+  return (m_id * (-1));
```

Comment:
May I presume that `m_id` starts at 1?  If it starts at 0, this will be problematic.

Reply:
This should return m_id, the conversion from the negative id to an array index should only happen internally to the make_device_allocator/get_device_allocator functions.

Reply:
And m_id should start at -1